enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,211 @@
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/website/)
+
+## Summary
+
+The proposal adds a new field `executorSubject` to the spec of `ManifestWork` 
+api, to support dynamic identity authentication/authorization upon the execution
+of `ManifestWork` resources. If the `executorSubject` is absent, it means the
+work agent will keep using the mounted service account to raise requests against
+the managed cluster which is also for preserving the backward compatibility.
+
+## Motivation
+
+### Separate the identity and its held permission for the work agent
+
+There can be different actors using the work api to distribute resource from
+the hub cluster. Currently the work agent has the cluster-admin access to the
+managed cluster which means that every hub users with write access to the
+work resources will be able to dispatch any resources to the managed cluster.
+This can be a security hole for OCM, e.g. one can apply/overwrite some
+cluster-scoped configuration to the managed cluster which potentially break/halt
+the cluster. Hence, we need a way to clarify the owner identity/role of the
+"ManifestWork" before it takes effect so that we can explicitly check whether 
+that owner has sufficient permission in the managed cluster.
+
+### Auditing the API request history from work agent
+
+Currently all the API requests raised by work agent in the managed cluster are
+from a fixed service account ("open-cluster-management-agent/klusterlet-work-sa"
+under the default installation). However in some cases, we need to clarify the
+source/owner identity of the `ManifestWork` resource so that we can record the
+api audits in a finer granularity. This will be helpful for the managed
+cluster's admin to be capable of knowing "who is operating that `ManifestWork`
+from the hub cluster".
+
+### Goals
+
+- `ManifestWork` resource be able to clarify the executor/owner identity before
+  actually writing the resources to the managed cluster.
+
+### Non-Goals
+
+- This proposal doesn't put any requirement to the selection of executor/owner in
+  the "ManifestWork", which allows hub admin to reference any valid Kubernetes 
+  user/role to be the executor/owner.
+
+## Proposal
+
+### API Changes
+
+This proposal adds a new field `executorSubject` to the `ManifestWork` api, 
+which is defaulted to nil if absent. A nil executor subject indicates that the 
+work agent doesn't perform any impersonation before requesting the hosting 
+managed cluster.
+
+
+A sample of the new `executorSubject` field in Yaml is shown below:
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata: ...
+spec:
+  executorSubject:
+    type: [ServiceAccount|Username] # union discriminator for extensibility
```

Comment:
I like the structure here, but I suggest starting with only serviceaccount.

Reply:
Users are short lived entities compared to serviceaccounts (they come and go from companies for instance).  Having this structure to choose a different subject type eventually is good, but I'd suggest extending it as part two.  Because you need compatibility with the existing API, you cannot start with a mechanism to default this to "me".

Reply:
@deads2k fair, i removed username type in the latest update